app/controllers/items_controller.rb

```diff
@@ -3,25 +3,30 @@
 class ItemsController < ApplicationController
   def index
     @items = current_organization.items.includes(:base_item, :kit).alphabetized.class_filter(filter_params)
+    @items = @items.active unless params[:include_inactive_items]
     @item_categories = current_organization.item_categories.includes(:items).order('name ASC')
     @kits = current_organization.kits.includes(line_items: :item, inventory_items: :storage_location)
     @storages = current_organization.storage_locations.active_locations.order(id: :asc)
 
     @include_inactive_items = params[:include_inactive_items]
     @selected_base_item = filter_params[:by_base_item]
-    @items_with_counts = ItemsByStorageCollectionQuery.new(organization: current_organization, filter_params: filter_params).call
-
-    @items_by_storage_collection_and_quantity = ItemsByStorageCollectionAndQuantityQuery.new(organization: current_organization, filter_params: filter_params).call
-    unless params[:include_inactive_items]
-      @items = @items.active
-      @items_with_counts = @items_with_counts.active
-    end
 
     @paginated_items = @items.page(params[:page])
 
+    if Event.read_events?(current_organization)
+      @inventory = View::Inventory.new(current_organization.id)
+    end
+    @items_by_storage_collection_and_quantity = ItemsByStorageCollectionAndQuantityQuery.call(organization: current_organization,
+      inventory: @inventory,
```

Comment:
Note to self: I verified that ItemsByStorageCollectionAndQuantityQuery handles nil inventory